// n8n-nodes-serpapi/dist/nodes/SerpApi/descriptions/FacebookProfileDescription.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.facebookProfileFields = void 0;
const SerpApiDescription_1 = require("./SerpApiDescription");
exports.facebookProfileFields = [
    {
        displayName: 'Profile ID (profile_id)',
        name: 'profile_id',
        description: 'Parameter defines the Facebook profile ID. You can find it in the URL of the profile page. For example, in https://www.facebook.com/Meta, the profile ID is Meta. In https://facebook.com/profile.php?id=100080376596424, the profile ID is 100080376596424.',
        default: 'Meta',
        routing: {
            request: {
                qs: {
                    profile_id: '={{$value}}'
                }
            }
        },
        type: 'string',
        required: true,
        displayOptions: {
            show: {
                operation: [
                    'facebook_profile'
                ]
            }
        }
    },
    {
        displayName: 'Additional Fields',
        name: 'additionalFields',
        type: 'collection',
        default: {},
        placeholder: 'Add Field',
        displayOptions: {
            show: {
                operation: [
                    'facebook_profile'
                ]
            }
        },
        options: [
            ...SerpApiDescription_1.serpApiFields
        ]
    }
];
