// n8n-nodes-serpapi/dist/nodes/SerpApi/descriptions/GoogleLocalDescription.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.googleLocalFields = void 0;
const SerpApiDescription_1 = require("./SerpApiDescription");
const GoogleOptions_1 = require("./GoogleOptions");
exports.googleLocalFields = [
    {
        displayName: 'Search Query (q)',
        name: 'q',
        description: 'Parameter defines the query you want to search',
        default: 'Coffee',
        routing: {
            request: {
                qs: {
                    q: '={{$value}}',
                },
            },
        },
        type: 'string',
        required: true,
        displayOptions: {
            show: {
                operation: ['google_local'],
            },
        },
    },
    {
        displayName: 'Location (location)',
        name: 'location',
        description: 'Parameter defines from where you want the search to originate. See docs for supported locations: https://serpapi.com/locations-api.',
        default: '',
        routing: {
            request: {
                qs: {
                    location: '={{$value}}',
                },
            },
        },
        type: 'string',
        displayOptions: {
            show: {
                operation: ['google_local'],
            },
        },
    },
    {
        displayName: 'Additional Fields',
        name: 'additionalFields',
        type: 'collection',
        default: {},
        placeholder: 'Add Field',
        displayOptions: {
            show: {
                operation: ['google_local'],
            },
        },
        options: [
            {
                displayName: 'Encoded Location (uule)',
                name: 'uule',
                description: "Parameter is the Google encoded location you want to use for the search. uule and location parameters can't be used together.",
                default: '',
                routing: {
                    request: {
                        qs: {
                            uule: '={{$value}}',
                        },
                    },
                },
                type: 'string',
            },
            {
                displayName: 'Domain (google_domain)',
                name: 'google_domain',
                description: 'Parameter defines the Google domain to use',
                default: 'google.com',
                routing: {
                    request: {
                        qs: {
                            google_domain: '={{$value}}',
                        },
                    },
                },
                type: 'options',
                options: GoogleOptions_1.DomainOptions,
            },
            {
                displayName: 'Country (gl)',
                name: 'gl',
                description: 'Parameter defines the country to use for the Google Local search',
                default: 'us',
                routing: {
                    request: {
                        qs: {
                            gl: '={{$value}}',
                        },
                    },
                },
                type: 'options',
                options: GoogleOptions_1.CountryOptions,
            },
            {
                displayName: 'Language (hl)',
                name: 'hl',
                description: 'Parameter defines the language to use for the search',
                default: 'en',
                routing: {
                    request: {
                        qs: {
                            hl: '={{$value}}',
                        },
                    },
                },
                type: 'options',
                options: GoogleOptions_1.LanguageOptions,
            },
            {
                displayName: 'Google Place ID (ludocid)',
                name: 'ludocid',
                description: 'Parameter defines the ID (CID) of the Google My Business listing you want to scrape. Also known as Google Place ID. See docs for more information: https://serpapi.com/google-local-api#api-parameters-advanced-google-local-parameters',
                default: '',
                routing: {
                    request: {
                        qs: {
                            ludocid: '={{$value}}',
                        },
                    },
                },
                type: 'string',
            },
            {
                displayName: 'Advanced Search Parameters (tbs)',
                name: 'tbs',
                description: "(to be searched) parameter defines advanced search parameters that aren't possible in the regular query field",
                default: '',
                routing: {
                    request: {
                        qs: {
                            tbs: '={{$value}}',
                        },
                    },
                },
                type: 'string',
            },
            {
                displayName: 'Result Offset (start)',
                name: 'start',
                description: 'Parameter defines the result offset. It skips the given number of results.',
                default: '',
                routing: {
                    request: {
                        qs: {
                            start: '={{$value}}',
                        },
                    },
                },
                type: 'string',
            },
            {
                displayName: 'Device (device)',
                name: 'device',
                description: 'Parameter defines the device to use to get the results. It can be set to desktop (default) to use a regular browser, tablet to use a tablet browser (currently using iPads), or mobile to use a mobile browser.',
                default: 'desktop',
                routing: {
                    request: {
                        qs: {
                            device: '={{$value}}',
                        },
                    },
                },
                type: 'string',
            },
            ...SerpApiDescription_1.serpApiFields,
        ],
    },
];
