// n8n-nodes-serpapi/dist/nodes/SerpApi/descriptions/GoogleAIOverviewFields.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.googleAIOverviewFields = void 0;
const SerpApiDescription_1 = require("./SerpApiDescription");
exports.googleAIOverviewFields = [
    {
        displayName: 'Search Query (page_token)',
        name: 'page_token',
        description: 'Parameter defines the token to fetch the AI Overview from. Requires an ai_overview.page_token from the AI Overview Results section of the Google Search API response. The page_token expires within 4 minutes of the search and must be used immediately.',
        default: null,
        routing: {
            request: {
                qs: {
                    page_token: '={{$value}}'
                }
            }
        },
        type: 'string',
        required: true,
        displayOptions: {
            show: {
                operation: [
                    'google_ai_overview'
                ]
            }
        }
    },
    {
        displayName: 'Additional Fields',
        name: 'additionalFields',
        type: 'collection',
        default: {},
        placeholder: 'Add Field',
        displayOptions: {
            show: {
                operation: [
                    'google_ai_overview'
                ]
            }
        },
        options: [
            ...SerpApiDescription_1.serpApiFields
        ]
    }
];
