// n8n-nodes-serpapi/dist/nodes/SerpApi/descriptions/YoutubeSearchDescription.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.youtubeSearchFields = void 0;
const SerpApiDescription_1 = require("./SerpApiDescription");
const GoogleOptions_1 = require("./GoogleOptions");
exports.youtubeSearchFields = [
    {
        displayName: 'Search Query (search_query)',
        name: 'search_query',
        description: 'Parameter defines the search query. You can use anything that you would use in a regular YouTube search.',
        default: 'Coffee',
        routing: {
            request: {
                qs: {
                    search_query: '={{$value}}'
                }
            }
        },
        type: 'string',
        required: true,
        displayOptions: {
            show: {
                operation: ['youtube']
            }
        }
    },
    {
        displayName: 'Additional Fields',
        name: 'additionalFields',
        type: 'collection',
        default: {},
        placeholder: 'Add Field',
        displayOptions: {
            show: {
                operation: ['youtube']
            }
        },
        options: [
            {
                displayName: 'Country (gl)',
                name: 'gl',
                description: 'Parameter defines the country to use for the YouTube search',
                default: 'us',
                routing: {
                    request: {
                        qs: {
                            gl: '={{$value}}',
                        },
                    },
                },
                type: 'options',
                options: GoogleOptions_1.CountryOptions,
            },
            {
                displayName: 'Language (hl)',
                name: 'hl',
                description: 'Parameter defines the language to use for the YouTube search',
                default: 'en',
                routing: {
                    request: {
                        qs: {
                            hl: '={{$value}}',
                        },
                    },
                },
                type: 'options',
                options: GoogleOptions_1.LanguageOptions,
            },
            {
                displayName: 'Filter (sp)',
                name: 'sp',
                description: 'Parameter can be used for pagination and filtering results. For pagination, use `next_page_token` in the `serpapi_pagination` object of the JSON response. For filters, visit the YouTube website, set filters you want and copy the `sp` value from the URL.',
                default: '',
                routing: {
                    request: {
                        qs: {
                            sp: '={{$value}}'
                        }
                    }
                },
                type: 'string'
            },
            ...SerpApiDescription_1.serpApiFields,
        ]
    }
];
